src/main/java/com/singh/astha/notification/service/controllers/NotificationController.java

```diff
@@ -0,0 +1,37 @@
+package com.singh.astha.notification.service.controllers;
+
+import com.singh.astha.notification.service.dtos.kafka.NotificationRequest;
+import com.singh.astha.notification.service.dtos.response.NotificationTokenResponseDto;
+import com.singh.astha.notification.service.dtos.response.wrapper.ResponseWrapper;
+import com.singh.astha.notification.service.services.NotificationService;
+import io.swagger.v3.oas.annotations.security.SecurityRequirement;
+import jakarta.validation.Valid;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/notification")
+@SecurityRequirement(name = "bearerAuth")
+public class NotificationController {
+
+    private final NotificationService notificationService;
+
+    public NotificationController(NotificationService notificationService) {
+        this.notificationService = notificationService;
+    }
+
+    @PostMapping()
+    public ResponseEntity<ResponseWrapper<NotificationTokenResponseDto>> sendNotification(
+            Authentication authentication,
```

Comment:
Why would there be User Authentication for this?

It should be Service to Service Authentication Mechanism instead of the User JWT Token. A user would never send himself/herself the notification.

Reply:
fixed